src/handlers/v1/markets/[chainId]/[marketId]/zap-supply/index.ts

```diff
@@ -0,0 +1,207 @@
+import { QuoteAPIResponseBody, ZapQuotation } from 'src/types';
+import * as apisdk from '@protocolink/api';
+import * as common from '@protocolink/common';
+import { utils } from 'ethers';
+import { validateMarket } from 'src/validations';
+
+type GetZapSupplyQuotationRouteParams = EventPathParameters<{ chainId: string; marketId: string }> &
+  EventBody<{
+    account?: string;
+    sourceToken?: common.TokenObject;
+    amount?: string;
+    targetToken?: common.TokenObject;
+    slippage?: number;
+  }>;
+
+type GetZapSupplyQuotationResponseBody = QuoteAPIResponseBody<ZapQuotation>;
+
+export const v1GetZapSupplyQuotationRoute: Route<GetZapSupplyQuotationRouteParams> = {
+  method: 'POST',
+  path: '/v1/markets/{chainId}/{marketId}/zap-supply',
+  handler: async (event) => {
+    const chainId = Number(event.pathParameters.chainId);
+    const marketId = event.pathParameters.marketId.toUpperCase();
+    if (!validateMarket(chainId, marketId)) {
+      throw newHttpError(400, { code: '400.1', message: 'market does not exist' });
+    }
+
+    if (!event.body) {
+      throw newHttpError(400, { code: '400.2', message: 'body is invalid' });
+    }
+
+    let account = event.body.account;
+    if (!account) {
+      throw newHttpError(400, { code: '400.3', message: `account can't be blank` });
+    }
+    try {
+      account = utils.getAddress(account);
+    } catch {
+      throw newHttpError(400, { code: '400.4', message: 'account is invalid' });
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR } = marketInfo;
+    const currentPosition = { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR };
+
+    let targetTokenAmount = '0';
+    const logics: GetZapSupplyQuotationResponseBody['logics'] = [];
+    let approvals: GetZapSupplyQuotationResponseBody['approvals'] = [];
+    let permitData: GetZapSupplyQuotationResponseBody['permitData'];
+    let targetPosition = currentPosition;
+    if (event.body.sourceToken && event.body.targetToken && event.body.amount && Number(event.body.amount) > 0) {
+      const { amount, slippage } = event.body;
+      const {
+        baseToken,
+        baseTokenPrice,
+        supplyAPR,
+        supplyUSD,
+        borrowAPR,
+        borrowCapacityUSD,
+        liquidationLimit,
+        collaterals,
+      } = marketInfo;
+      const sourceToken = common.Token.from(event.body.sourceToken);
+      const targetToken = common.Token.from(event.body.targetToken);
+
+      // 1. check supply collateral or base token
+      const targetCollateral = collaterals.find(({ asset }) => asset.is(targetToken.unwrapped));
+      if (!targetCollateral && !targetToken.is(baseToken)) {
```

Comment:
```suggestion
      if (!targetCollateral && !targetToken.unwrapped.is(baseToken)) {
```
With the support for both wrapped and unwrapped tokens for `targetToken` this [commit](https://github.com/dinngo/compound-kit-api/commit/a8fb589070379266a4693b790946564488b2b32d) for various operations, please review all instances of `targetToken` usage.